### ISSUE TITLE
When template folder is present, changelog_file is ignored

### ISSUE THREAD

**Issue by GiamBoscaro** [python-semantic-release/python-semantic-release#983]
## The problem

If I create a custom template for the CHANGELOG and put it into the templates directory, the parameter `changelog_file` is ignored and the CHANGELOG is always generated in the root folder of the project with the name `CHANGELOG.md`.
This happens if I specify or also if I do not specify `template_dir`.

## Expected behavior

CHANGELOG should be generated in the location specified in `changelog_file`.

## Environment

Please state which OS you are using and provide the output of the following commands:

```shell
python --version
# Python 3.11.2
pip --version
# pip 23.0.1 from /usr/lib/python3/dist-packages/pip (python 3.11)
semantic-release --version
# semantic-release, version 9.8.5
pip freeze
# nothing
```

## Configuration

```toml
[tool.semantic_release]
assets = ['docs/CHANGELOG.md', 'src/semantic_versioning_sample/__init__.py']
commit_author = 'semantic-release-bot <semantic-release-bot@mail.com>'
commit_message = 'chore(release): {version} [skip ci]'
commit_parser = 'angular'
version_variables = ['src/semantic_versioning_sample/__init__.py:VERSION']
tag_format = "v{version}"
allow_zero_version = false

[tool.semantic_release.remote]
type = 'bitbucket'
ignore_token_for_push = true
insecure = false

[tool.semantic_release.changelog]
changelog_file = 'docs/CHANGELOG.md'
template_dir = 'templates'
exclude_commit_patterns = ["build", "chore", "ci", "style", "refactor", "test", "Pull request", "Merge"]
    
[tool.semantic_release.branches.main]
match = "(main|master)"
prerelease = false

[tool.semantic_release.branches.stage]
match = "(stage|qa)"
prerelease_token = "rc"
prerelease = true
```

## Logs

I cannot post the whole log is way too long. But here you can see that it is saving the `CHANGELOG.md` into the wrong folder, ignoring what is written in the `toml` file.

```shell
semantic-release -vv changelog

 INFO     [semantic_release.changelog.template] INFO template.recursive_render:         template.py:100
                    Rendering templates from                                                                     
                    /workspaces/semantic-versioning-sample-python/templates to                                   
                    /workspaces/semantic-versioning-sample-python                                                
 DEBUG    [semantic_release.changelog.template] DEBUG template.recursive_render:        template.py:110
          rendering CHANGELOG.md.j2 to                                                                 
          /workspaces/semantic-versioning-sample-python/CHANGELOG.md      
```

**Comment by codejedi365** [comment]
@GiamBoscaro This is expected behavior. It's one or the other. If you want a specific location for the changelog just mimic the file structure inside of the templates directory. See #923.

Please also search the issues before opening another as this was previously asked, duplicate of #923.

**Comment by GiamBoscaro** [comment]
OK understood, thank you! Should be written in the documentation to make it more clear maybe. In the configuration section, on the `changelog_file` parameters there could be a note like "ignored if templates_dir is set or default templates folder is present"

**Comment by codejedi365** [comment]
Agreed. I likely meant to do this previously. The docs have a lot of improvements to go

**Comment by codejedi365** [comment]
Docs were updated.

**Comment by codejedi365** [comment]
### :tada: This resolution has been included in version 9.8.7 :tada:

The release is available on:
- [PyPi package](https://pypi.org/project/python-semantic-release/9.8.7/)
- [GitHub release](https://github.com/python-semantic-release/python-semantic-release/releases/tag/v9.8.7)
